gempy/core/solution.py

```diff
@@ -275,43 +257,18 @@ def compute_all_surfaces(self, **kwargs):
         series_type = self.series.df['BottomRelation']
         s_n = 0
         active_indices = self.surfaces.df.groupby('isActive').groups[True]
-        if 'rescale' in kwargs:
-            rescale = kwargs.pop('rescale')
-        else:
-            rescale = False
+        rescale = kwargs.pop('rescale', False)
 
         # We loop the scalar fields
         for e, scalar_field in enumerate(self.scalar_field_matrix):
             sfas = self.scalar_field_at_surface_points[e]
             # Drop
             sfas = sfas[np.nonzero(sfas)]
-
-            # TODO: I think this condition is not necessary anymore
-
-            # #elif series_type[e] == 'Fault':
-            # #    mask_array = None
-            # else:
-            #     mask_array = self.mask_matrix_pad[e]
-            # if series_type[e] == 'Fault':
-            #     mask_array = np.invert(self.mask_matrix_pad[e])
-            # else:
-            #     mask_array = self.mask_matrix_pad[e]
-            if series_type[e-1] == 'Onlap':
-                mask_array = self.mask_matrix_pad[e-1]
-            else:
-                mask_array = self.mask_matrix_pad[e]
-
+            mask_array = self.mask_matrix_pad[e-1 if series_type[e-1] == 'Onlap' else e]
```

Comment:
```suggestion
            mask_array = self.mask_matrix_pad[e-1 if series_type[e-1] == 'Onlap' else self.mask_matrix_pad[e]
```
or simply doing the condition for the index and keep `mask_array = self.mask_matrix_pad[aux_index]`

Reply:
That suggestion is incorrect, the original if statement was
```python
if series_type[e-1] == 'Onlap':
       mask_array = self.mask_matrix_pad[e-1]
else:
       mask_array = self.mask_matrix_pad[e]
```

so mask_array equal either the values at the index "e" or "e-1" in mask_matrix_pad. 

```python
#my code
mask_array = self.mask_matrix_pad[e-1 if if series_type[e-1] == 'Onlap' else e]
```
This suggestion would try to index mask_matrix_pad with the value at mask_matrix_pad[e] which would fail if the datatype is not an integer. Anyways I don't think that suggestion is logically what the original if statement did.

Reply:
ah yes now I see the squared brackets sorry! First time I see that pattern. Pretty cool :D Alright lets see if I fix travis today and I will merge it. Thanks again!